packages/ui-color-picker/src/ColorIndicator/index.tsx

```diff
@@ -0,0 +1,111 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/** @jsx jsx */
+import { Component } from 'react'
+
+import { testable } from '@instructure/ui-testable'
+import { passthroughProps } from '@instructure/ui-react-utils'
+import { withStyle, jsx } from '@instructure/emotion'
+import { hexToRgb } from '@instructure/ui-color-utils/src/conversions'
+import { isValid } from '@instructure/ui-color-utils/src/isValid'
+
+import generateStyle from './styles'
+import generateComponentTheme, { colorIndicatorBorderColor } from './theme'
+
+import type { ColorIndicatorProps } from './props'
+
+type RGBAType = {
+  r: number
+  g: number
+  b: number
+  a: number
+}
+
+/**
+---
+category: components
+---
+@tsProps
+**/
+@withStyle(generateStyle, generateComponentTheme)
+@testable()
+class ColorIndicator extends Component<ColorIndicatorProps> {
+  static readonly componentId = 'ColorIndicator'
+  static defaultProps = {
+    onClick: null,
+    disabled: false
+  }
+  ref: Element | null = null
+
+  componentDidMount() {
+    this.props.makeStyles?.()
+  }
+
+  componentDidUpdate() {
+    this.props.makeStyles?.()
+  }
+
+  handleRef = (el: Element | null) => {
+    const { elementRef } = this.props
+
+    this.ref = el
+
+    if (typeof elementRef === 'function') {
+      elementRef(el)
+    }
+  }
+
+  calcBlendedColor = (c1: RGBAType, c2: RGBAType) => {
+    // 0.4 as decided by design
+    const c2Alpha = c2.a * 0.4
+    const c1Alpha = 1 - c2Alpha
+    const alpha = 1 - c1Alpha * (1 - c1Alpha)
+
+    return `rgba(
+      ${(c2.r * c2Alpha) / alpha + (c1.r * c1Alpha * (1 - c2Alpha)) / alpha},
+      ${(c2.g * c2Alpha) / alpha + (c1.g * c1Alpha * (1 - c2Alpha)) / alpha},
+      ${(c2.b * c2Alpha) / alpha + (c1.b * c1Alpha * (1 - c2Alpha)) / alpha},
+      ${c2.a})`
+  }
+  render() {
+    const { color, elementRef, ...props } = this.props
+
+    return (
+      <div
+        {...passthroughProps(props)}
+        css={this.props.styles?.checkerBoardWithColor}
+        style={{
+          borderColor: this.calcBlendedColor(
+            hexToRgb(colorIndicatorBorderColor),
+            hexToRgb(isValid(color || 'none') ? color || 'none' : '#fff')
```

Comment:
I think this can be simplified: `tinyColor`'s `Color()` allows undefined values (you need to allow this in our `isvalid` method), so you can just write `  hexToRgb(isValid(color) ? color : '#fff')`

Reply:
I remember that there was an edge-case where I had to use this logic, but I can't remember what was it, nor can I reproduce, so for the time being I removed it. Maybe it was fixed later, but I never thought about it